### PR TITLE
Updating required node version

### LINKED ISSUE #5178: Node JS version too tightly defined
The nodejs version in `package.json` is locked to `~4.2.0` which is too tightly coupled when `^4.2.0` should be fine.

Following the [installation guidelines on nodejs.org](https://nodejs.org/en/download/package-manager/#enterprise-linux-and-fedora) ends up installing `4.4.0` which *should* be fine.

Locking to `4.2.x` makes contributing unnecessarily difficult

### PR DESCRIPTION
**Pull request by dhensby**
Fixes #5178

### CODE REVIEW

**Comment by mikenz** [comment]
+1
Thanks :-)

File: package.json [hunk 1]
```diff
@@ -7,10 +7,12 @@
     "test": "tests"
   },
   "engines": {
-    "node": "~4.2.0"
+    "node": "^4.2.0"
   },
   "scripts": {
+    "gulp": "gulp",
```
Comment (flashbackzoo) [inline package.json]: This will error - there's no 'default' task in our gulp.

File: package.json [hunk 2]
```diff
@@ -7,10 +7,12 @@
...
     "build": "gulp build",
+    "bundle": "gulp bundle",
```
Comment (flashbackzoo) [inline package.json]: Can you please write some docs for this? The are others in `docs/en/05_Contributing/01_Code.md`

File: package.json [hunk 3]
```diff
@@ -7,10 +7,12 @@
     "test": "tests"
   },
   "engines": {
-    "node": "~4.2.0"
+    "node": "^4.2.0"
```
Comment (flashbackzoo) [inline package.json]: :+1:

Comment (chillu) [inline package.json]: Ah yep, we've done that in the `integrate-react` branch already, but never merged it back.

File: package.json (hunk 2 above)
Comment (dhensby) [inline package.json]: If you give me the words, I'll add them :P

File: docs/en/05_Contributing/01_Code.md [hunk 4]
```diff
@@ -278,6 +278,13 @@ The [Node.js](https://nodejs.org) JavaScript runtime is the foundation of our cl
 The configuration for an npm package goes in `package.json`. You'll see one in the root directory of `framework`. As well as being used for defining dependencies and basic package information, the `package.json` file has some handy scripts.
 
 ```
+$ npm run bundle
+```
+
+This script comiles the SCSS in CSS, the javascript source files into their dist and, minifies and concatinates all the
```
Comment (flashbackzoo) [inline docs/en/05_Contributing/01_Code.md]: This doesn't do anything with CSS.

"This script generates the `lib` and `leftandmain` JavaScript bundles"

Comment (dhensby) [inline docs/en/05_Contributing/01_Code.md]: fixed

**Comment by dhensby** [comment]
thanks

**Comment by tractorcow** [comment]
Node problem

**Pull request merged**